src/librustc_metadata/creader.rs

```diff
@@ -853,6 +853,33 @@ impl<'a> CrateLoader<'a> {
         }
     }
 
+    fn inject_profiler_runtime(&mut self) {
+        if self.sess.opts.debugging_opts.profile {
+            let mut uses_std = false;
+            self.cstore.iter_crate_data(|_, data| {
```

Comment:
@japaric can you remind me why this is needed?

Naively I'd expect that a crate would require the profiler runtime regardless of whether it links to std or not, but I forget why this clause is above.